Tests/FoundationTests/NotificationCenterExtensionsTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  NotificationCenterExtensionsTests.swift
+//  SwifterSwift
+//
+//  Created by Guy Kogus on 6/3/20.
+//  Copyright © 2020 SwifterSwift
+//
+
+import XCTest
+@testable import SwifterSwift
+
+final class NotificationCenterExtensionsTests: XCTestCase {
+
+    func testObserveOnce() {
+        var count = 0
+
+        let notificationCenter = NotificationCenter()
+        let notificationName = Notification.Name(rawValue: "foo")
+        let operationQueue = OperationQueue()
+        let increment = { (_: Notification) in count += 1 }
+        notificationCenter.observeOnce(forName: notificationName,
+                                       object: self,
+                                       queue: operationQueue,
+                                       using: increment)
+
+        let wrongNotificationName = Notification.Name(rawValue: "bar")
+
+        notificationCenter.post(name: wrongNotificationName, object: self)
+        XCTAssertEqual(count, 0)
+        notificationCenter.post(name: notificationName, object: nil)
+        XCTAssertEqual(count, 0)
+        notificationCenter.post(name: notificationName, object: self)
+        XCTAssertEqual(count, 1)
+        notificationCenter.post(name: notificationName, object: self)
+        XCTAssertEqual(count, 1)
+
+        notificationCenter.observeOnce(forName: nil,
+                                       object: self,
```

Comment:
Ahh I think the Linux is failing may be because you are passing `self` here? And in fact other places which maybe is not necessary ... because it only needs the closure

Reply:
And sorry for the delay @guykogus, I still don't know how did I miss your message 😅

Reply:
Nice catch! Such a strange bug... the Linux Swift implementation is such a PITA!

Reply:
Yeah, there are differences that we really not expect to have from Darwin to Linux 🙃